librz/core/cmd/cmd_analysis.c

```diff
@@ -6333,6 +6033,254 @@ RZ_IPI RzCmdStatus rz_il_vm_status_handler(RzCore *core, int argc, const char **
+			break;
+		case 'J':
+		case 'j':
+			rz_core_analysis_callgraph(core, UT64_MAX, RZ_GRAPH_FORMAT_JSON);
+			break;
+		case 'g':
+			rz_core_analysis_callgraph(core, UT64_MAX, RZ_GRAPH_FORMAT_GML);
+			break;
+		case 'd':
+			rz_core_analysis_callgraph(core, UT64_MAX, RZ_GRAPH_FORMAT_DOT);
+			break;
+		case '*':
+			rz_core_analysis_callgraph(core, UT64_MAX, RZ_GRAPH_FORMAT_CMD);
+			break;
+		default:
+			rz_warn_if_reached();
+			break;
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_graph_diff_handler(RzCore *core, int argc, const char **argv) {
+	int diff_opt = RZ_CORE_ANALYSIS_GRAPHBODY | RZ_CORE_ANALYSIS_GRAPHDIFF;
+	char format = argc > 1 ? argv[1][0] : ' ';
+	ut64 addr = argc > 2 ? rz_num_math(core->num, argv[2]) : core->offset;
+	switch (format) {
+	case 'j': {
+		rz_core_gdiff_function_1_file(core, addr, core->offset);
+		rz_core_analysis_graph(core, addr, diff_opt | RZ_CORE_ANALYSIS_JSON);
+		break;
+	}
+	case 'J': {
+		rz_core_gdiff_function_1_file(core, addr, core->offset);
+		rz_core_analysis_graph(core, addr, diff_opt | RZ_CORE_ANALYSIS_JSON | RZ_CORE_ANALYSIS_JSON_FORMAT_DISASM);
+		break;
+	}
+	case '*': {
+		rz_core_gdiff_function_1_file(core, addr, core->offset);
+		rz_core_analysis_graph(core, addr, diff_opt | RZ_CORE_ANALYSIS_STAR);
+		break;
+	}
+	case ' ':
+	case 't':
+	case 'k':
+	case 'v':
+	case 'g': {
+		rz_core_agraph_reset(core);
+		rz_core_cmdf(core, ".agd* @ %" PFMT64u "; agg%s;", addr, input + 1);
+		break;
+	}
+	case 'd': {
+		rz_core_gdiff_function_1_file(core, addr, core->offset);
+		rz_core_analysis_graph(core, addr, diff_opt);
+		break;
+	}
+	case 'w': {
+		char *cmdargs = rz_str_newf("agdd 0x%" PFMT64x, core->offset);
+		convert_dotcmd_to_image(core, cmdargs, input + 2);
+		free(cmdargs);
+		break;
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_graph_bb_function_handler(RzCore *core, int argc, const char **argv) {
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_graph_imports_handler(RzCore *core, int argc, const char **argv) {
+	RzGraph *graph = rz_core_analysis_importxrefs(core);
+	if (!graph) {
+		RZ_LOG_ERROR("Couldn't create graph\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_core_graph_print(core, graph, -1, true, input + 1);
+	rz_graph_free(graph);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_graph_refs_handler(RzCore *core, int argc, const char **argv) {
+	char format = argc > 2 : argv[1][0] : ' ';
+	if (format == '*') {
+		rz_core_analysis_coderefs(core, core->offset);
+	} else {
+		core->graph->is_callgraph = true;
+		rz_core_agraph_reset(core);
+		rz_core_cmdf(core, ".agr* @ %" PFMT64u ";", core->offset);
+		rz_core_agraph_print(core, -1, input + 1);
+		core->graph->is_callgraph = false;
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_graph_refs_global_handler(RzCore *core, int argc, const char **argv) {
+	char format = argc > 2 : argv[1][0] : ' ';
+	if (format == '*') {
```

Comment:
let's use the strings and not the chars. Otherwise we risk recognizing stuff like `*adasdsadas` just because the first char is `*`. It should be an error if it doesn't match exactly what we want.